### PR TITLE
Add strict?-option to Ash.Query.load

### PR DESCRIPTION
**Pull request by Torkan**
Enables more optimised data-fetching by only fetching the specified attributes of loaded/nested relationships.

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [x] Documentation changes
- [x] Features include unit/acceptance tests
- [x] Refactoring
- [ ] Update dependencies

### CODE REVIEW

File: test/actions/load_test.exs [hunk 1]
```diff
@@ -1608,5 +1608,106 @@ defmodule Ash.Test.Actions.LoadTest do
...
+               |> Ash.read!()
+
+      [loaded_post] = author.posts
+
+      assert post.contents == "post content"
+      assert loaded_post.title == "author post"
+      assert %Ash.NotLoaded{} = loaded_post.contents
+    end
+
+    @tag :new
```
Comment (zachdaniel) [inline test/actions/load_test.exs:1639]:
```suggestion
```

File: test/actions/load_test.exs [hunk 2]
```diff
@@ -1608,5 +1608,106 @@ defmodule Ash.Test.Actions.LoadTest do
...
+               |> Ash.Query.load([:posts], strict?: true)
+               |> Ash.read!()
+
+      [loaded_post] = author.posts
+
+      assert post.title == loaded_post.title
+      assert post.contents == loaded_post.contents
+    end
+
+    @tag :new
```
Comment (zachdaniel) [inline test/actions/load_test.exs:1666]:
```suggestion
```

File: test/actions/load_test.exs [hunk 3]
```diff
@@ -1608,5 +1608,106 @@ defmodule Ash.Test.Actions.LoadTest do
...
+
+      loaded_author = Ash.load!(author, [posts: [:title]], strict?: true)
+      [loaded_post] = loaded_author.posts
+
+      assert post.contents == "post content"
+      assert loaded_post.title == "author post"
+      assert %Ash.NotLoaded{} = loaded_post.contents
+    end
+
+    @tag :new
```
Comment (zachdaniel) [inline test/actions/load_test.exs:1690]:
```suggestion
```

**Review by zachdaniel — APPROVED** [review]
LGTM 🥳 Just had some small cosmetic change requests

**Comment by Torkan** [comment]
@tag entries in tests removed 😅

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**